platform/mellanox/cmis_host_mgmt/cmis_host_mgmt.py

```diff
@@ -0,0 +1,151 @@
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import shutil
+import click
+import re
+import os
+import subprocess
+
+
+class CMISHostMgmtActivator:
+    PARAMS = {
+        "sai_profile": {
+            "file_name": "sai.profile",
+            "enabled_param": "SAI_INDEPENDENT_MODULE_MODE=1",
+            "disabled_param": "SAI_INDEPENDENT_MODULE_MODE=0"
+        },
+        "pmon_daemon_control": {
+            "file_name": "pmon_daemon_control.json",
+            "enabled_param": "\"skip_xcvrd_cmis_mgr\": false",
+            "disabled_param": "\"skip_xcvrd_cmis_mgr\": true",
+        },
+        "sai_xml": {
+            "file_name": "sai_<>.xml", # will be filled at main, since we can't know the SKU here 
+            "enabled_param": "<late-create-all-ports>1</late-create-all-ports>",
+            "disabled_param": "<late-create-all-ports>1</late-create-all-ports>" # Shouldn't be called
+        }
+    }
+
+    @staticmethod
+    def change_param(param, path, action):
+        file_path = '{}/{}'.format(path, CMISHostMgmtActivator.PARAMS[param]["file_name"])
+        lines = None
+
+        with open(file_path, 'r') as param_file:
+            lines = param_file.read()
+
+        if lines:
+            if action == "disable":
+                lines = re.sub(CMISHostMgmtActivator.PARAMS[param]["enabled_param"],
+                               CMISHostMgmtActivator.PARAMS[param]["disabled_param"],
+                               lines)
+            elif action == "enable":
+                if param == "sai_profile" and not re.search(CMISHostMgmtActivator.PARAMS[param]["disabled_param"], lines):
+                    with open(file_path, 'a') as param_file:
+                        param_file.write(CMISHostMgmtActivator.PARAMS[param]["enabled_param"])
+                    return
+
+                lines = re.sub(CMISHostMgmtActivator.PARAMS[param]["disabled_param"],
+                               CMISHostMgmtActivator.PARAMS[param]["enabled_param"],
+                               lines)
+
+        with open(file_path, 'w') as param_file:
+            param_file.write(lines)
+
+
+    @staticmethod
+    def parse_show_platform_summary():
+        summary = subprocess.check_output(['show', 'platform', 'summary'])
+        summary = summary.decode('utf-8')
+        summary = [x for x in summary.split('\n') if x]
+
+        for field in summary:
+            key, value = field.split(": ")
+            
+            if key == 'Platform':
+                platform = value
+
+            elif key == 'HwSKU':
+                sku = value
+
+        return platform, sku
+
+
+    @staticmethod
+    def remove_im_file(file_path):
```

Comment:
"im" stands for "independent module"?
maybe change it just to remove_file/copy_file?

Reply:
Changed